cli/src/cmd/bind.rs

```diff
@@ -0,0 +1,184 @@
+use crate::cmd::Cmd;
+
+use clap::{Parser, ValueHint};
+use ethers::contract::MultiAbigen;
+use foundry_config::{
+    figment::{
+        self,
+        error::Kind::InvalidType,
+        value::{Dict, Map, Value},
+        Metadata, Profile, Provider,
+    },
+    impl_figment_convert, Config,
+};
+use serde::Serialize;
+use std::{fs, path::PathBuf};
+
+impl_figment_convert!(BindArgs);
+
+#[derive(Debug, Clone, Parser, Serialize)]
+pub struct BindArgs {
+    #[clap(
+        help = "The project's root path. By default, this is the root directory of the current Git repository or the current working directory if it is not part of a Git repository",
+        long,
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(skip)]
+    pub root: Option<PathBuf>,
+
+    #[clap(
+        help = "Path to where the contract artifacts are stored",
+        long = "out",
+        short,
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(rename = "out", skip_serializing_if = "Option::is_none")]
+    pub out_path: Option<PathBuf>,
+
+    #[clap(
+        help = "A directory at which to write the generated rust crate",
+        long = "bindings-root",
+        value_hint = ValueHint::DirPath
+    )]
+    #[serde(skip)]
+    pub bindings_root: Option<PathBuf>,
+
+    #[clap(
+        long = "crate-name",
+        help = "The name of the rust crate to generate. This should be a valid crates.io crate name. However, it is not currently validated by this command."
+    )]
+    #[serde(skip)]
+    crate_name: Option<String>,
+
+    #[clap(
+        long = "crate-version",
+        help = "The version of the rust crate to generate. This should be a standard semver version string. However, it is not currently validated by this command."
+    )]
+    #[serde(skip)]
+    crate_version: Option<String>,
+
+    #[clap(
+        long = "overwrite",
+        help = "Overwrite existing generated bindings. If set to false, the command will check that the bindings are correct, and then exit. If set to true, it will instead delete and overwrite the bindings."
+    )]
+    #[serde(skip)]
+    overwrite: bool,
+
+    #[clap(long = "single-file", help = "Generate bindings as a single file.")]
+    #[serde(skip)]
+    single_file: bool,
+}
+
+impl BindArgs {
+    /// Get the path to the foundry artifacts directory
+    fn artifacts(&self) -> PathBuf {
+        let c: Config = self.into();
+        c.out
+    }
+
+    /// Get the path to the root of the autogenerated crate
+    fn bindings_root(&self) -> PathBuf {
+        self.bindings_root.clone().unwrap_or_else(|| self.artifacts().join("bindings"))
+    }
+
+    /// Get the name of the autogenerated crate
+    fn crate_name(&self) -> String {
+        self.crate_name.clone().unwrap_or_else(|| "foundry-contracts".to_owned())
```

Comment:
Think you can instead use `default_value = "foundry-contracts"`  in the clap definition and remove the `Option`?